### PR TITLE
Issue #2572119: Converting evaluate() to doEvaluate() in EntityIsOfTy…

### PR DESCRIPTION
**Pull request by emclaughlin1215**
…pe condition.

See d.o issue https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/Condition/EntityIsOfType.php [hunk 1]
```diff
@@ -33,11 +34,17 @@
...
+   * @param string $specified_type
+   *   The type to check for
+   *
+   * @return bool
+   *   TRUE if the entity is of the provided type.
    */
-  public function evaluate() {
-    $provided_entity = $this->getContextValue('entity');
-    $specified_type = $this->getContextValue('type');
+  protected function doEvaluate(EntityInterface $provided_entity, $specified_type) {
```
Comment (klausi) [inline src/Plugin/Condition/EntityIsOfType.php]: Let's use the same variable names as in the annotation.

**Comment by emclaughlin1215** [comment]
Addressed the comments and pushed up changes.

File: src/Plugin/Condition/EntityIsOfType.php [hunk 2]
```diff
@@ -33,15 +34,21 @@
 class EntityIsOfType extends RulesConditionBase {
 
   /**
-   * {@inheritdoc}
+   * Check if the provided entity is of a specific type.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to check for a type
```
Comment (klausi) [inline src/Plugin/Condition/EntityIsOfType.php]: missing dot at the end.

File: src/Plugin/Condition/EntityIsOfType.php [hunk 3]
```diff
@@ -33,15 +34,21 @@
...
+   * @param string $type
+   *   The type to check for
```
Comment (klausi) [inline src/Plugin/Condition/EntityIsOfType.php]: missing dot at the end.

**Comment by emclaughlin1215** [comment]
Updated per comments.

**Comment by klausi** [comment]
merged, thanks!

**Pull request closed without merging**